### ISSUE TITLE
Scrollbar design tokens

### ISSUE THREAD

**Issue by jmanke** [Esri/calcite-design-system#8621]
**monday.com sync:** #9485786597

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

Currently, there is no way to customize the scrollbar style of components. In Dashboards, we have a custom scrollbar style that is globally applied. As we integrated calcite components, we observe differences between the default scrollbar styles of calcite-components, and our scrollbar styles.

We would like to control scrollbar styles as some OS + Browser's don't have visually appealing scrollbars, or they conflict with the theme the Dashboard is going for. For example, a dark themed Dashboard could have a very light colored scrollbar in Windows browsers.

cc @ffaubry @jcfranco @vijayendranj

### Acceptance Criteria

A way to customize components' scrollbar styles (when they are in the shadow root) - the first thought is to use design tokens (CSS variables). 

### Which Component

Any component that has a scrollbar inside of the shadow root.

### Example Use Case

Dashboards look beautiful, cohesive, and focused. Scroll bars play into that. When they don't look right (too big, not right color, different across elements), they detract from those qualities. Being able to apply scrollbars styles to all relevant `calcite-components` will ensure the application's style is consistent and aligns with our styling goals.

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [X] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Dashboards

**Comment by jcfranco** [comment]
Discussed this with the team and we would need to wait for [`scrollbar-width`](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width) and [`scrollbar-color`](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color) to be widely supported across browsers for this. Safari is the only one lacking support, but it looks that might change soon based on the following WebKit PRs: https://github.com/WebKit/WebKit/pull/14597, https://github.com/WebKit/WebKit/pull/13992.

Testing on browsers that support this show these styles piercing through shadow DOM: https://codepen.io/jcfranco/pen/OJqOvgG.

**Comment by vijayendranj** [comment]
//cc @ffaubry This is something web components team may also be interested.

**Comment by geospatialem** [comment]
For design to consider implementation.

_Of note Safari would not have support as mentioned in [Franco's comment above](https://github.com/Esri/calcite-design-system/issues/8621#issuecomment-1930996541)._

**Comment by jcfranco** [comment]
`scrollbar-width` is now supported in Safari. 🎉 `scrollbar-color` is still unsupported ([limitation noted](https://bugs.webkit.org/show_bug.cgi?id=231590#c3) in bug ticket).

I think we could proceed with this one by following the fallback pattern suggested [here](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar#adding_a_fallback_for_scrollbar_styles) along with overridable CSS props (a la `--calcite-color-focus`) or internal ones, as @macandcheese suggested.

> maybe we could expose some non-public css variables here for Esri teams that want to adopt despite the limited browser support?

@macandcheese Were you thinking these would be scoped to all components or would they also apply to the whole page?

**Comment by matgalla** [comment]
Spoke with the @Esri/calcite-designers and agreed that the default CSS property [`scrollbar-color`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-color) is sufficient in liu of a redundant semantic token. It can be scoped to apply the changes globally and/or within components in the major OS+browser combos:
<img width="2280" alt="Image" src="https://github.com/user-attachments/assets/4cbdd669-4218-41d2-af8c-850ff3881b6e" />
 
Codepen using semantic color tokens: https://codepen.io/matgalla/pen/ZYBzYdm

@Esri/calcite-pes I think we can close this issue since `scrollbar-color` is functional. If there is a need for [`scrollbar-gutter`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-gutter) or [`scrollbar-width`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-width) to take effect within components then we can either open a new issue or re-open/re-scope this one. Thoughts?

**Comment by DintaMel** [comment]
> @Esri/calcite-pes I think we can close this issue since scrollbar-color is functional. If there is a need for [scrollbar-gutter](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-gutter) or [scrollbar-width](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-width) to take effect within components then we can either open a new issue or re-open/re-scope this one. Thoughts?

@matgalla My understanding of the original issue is that `scrollbar-width` is also being requested.

> Dashboards look beautiful, cohesive, and focused. Scroll bars play into that. When they don't look right (**too big**, not right color, different across elements), they detract from those qualities.

**Comment by ashetland** [comment]
Unless I'm mistaken, our components don't render scrollbars themselves. I think native CSS may be the solve here.

**Comment by matgalla** [comment]
@DintaMel Good call. I think the action item here is to (potentially) change whatever is preventing components from accepting the modifications coming from `scrollbar-width` then. You can see in my codepen that the page-level scrollbar is impacted by the `thin` value, but the one within Panel is not.

**Comment by ellenupp** [comment]
Here is some more context for this ask:

In Dashboards, any elements where we control the scrolling will have a styled scrollbar. Here you can see one in the slide out sidebar on the left, and the list element ("Descriptions") on the right side. 

<img width="800" height="525" alt="Image" src="https://github.com/user-attachments/assets/76544829-b371-45b0-a304-65b05ad3697c" />
 <br>

Wherever scrolling is controlled by a calcite component the scrollbars won't be styled. For example in the map popup (calcite-flow) or in a dialog. [screenshot below shows default scrollbar on Mac in the map popup]

<img width="672" height="800" alt="Image" src="https://github.com/user-attachments/assets/1f71cbb7-cce9-4eea-bd09-aad05bf0d7a1" />
 <br>

To add another variation, arcgis-charts also has different scrollbars. 

<img width="300" height="244" alt="Image" src="https://github.com/user-attachments/assets/a37e5822-b1ec-452a-b7a4-ebcf2f2b8a47" />
 <br>

We’d like a way to have uniform scrollbars throughout the app, as there are currently three different ones (dashboard styled, default in calcite, and charts). We style scrollbars in Dashboards as the default ones on Windows are quite cumbersome and can detract from the visual appeal of the dashboard. The screenshot below shows the 3 different scrollbars on Windows Chrome. One of the goals was the reduce the width of it, as you can see the styled one on the left is thinner than the default one on the right. We’ve set the width (and height for horizontal ones) to 7px.

<img width="800" height="393" alt="Image" src="https://github.com/user-attachments/assets/24d81a12-a687-43b5-8453-c135b200caba" />
 <br>

Unfortunately this isn’t a perfect solution as the height/width properties don’t have an effect in Windows Firefox.  

<img width="800" height="451" alt="Image" src="https://github.com/user-attachments/assets/dc27006a-9a1b-4121-82b9-d192cfb567ac" />
 <br>

We also set the scrollbar thumb and background colors to either a light or dark version to match the theme. Here is one in a dark theme.

<img width="471" height="572" alt="Image" src="https://github.com/user-attachments/assets/189489d7-50e1-49b3-b5bb-738f91632c24" />
<br>

This is the style we use:
```
::-webkit-scrollbar {
  height: 7px;
  width: 7px;
}
::-webkit-scrollbar-corner {
  background: var(--calcite-color-foreground-1);
}
::-webkit-scrollbar {
  background: var(--scrollbar-background, #dfdfdf); /* border-3 */
}
::-webkit-scrollbar-thumb {
  background: var(--scrollbar-thumb-background, #cacaca); /* border-1 */
}
::-webkit-scrollbar-thumb:hover {
  background: var(--scrollbar-thumb-background-hover, #949494); /* border-input */
}
```

**Comment by DintaMel** [comment]
@matgalla I was going to add the styles Ellen mentioned to our acceptance criteria, but the styles are pseudo-elements, which i don't recall seeing other similar tokens for. Would we still stick with `scrollbar-color` and `scrollbar-width` as planned? See [MDN docs on `::-webkit-scrollbar`](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/::-webkit-scrollbar) for reference.

**Comment by jcfranco** [comment]
> [@DintaMel](https://github.com/DintaMel) Good call. I think the action item here is to (potentially) change whatever is preventing components from accepting the modifications coming from `scrollbar-width` then. You can see in my codepen that the page-level scrollbar is impacted by the `thin` value, but the one within Panel is not.

That's because [`scrollbar-color` inherits](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-color#formal_definition) by default, while [`scrollbar-width` does not](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/scrollbar-width#formal_definition). We'll need to update component styles to allow this (inherit in host and propagate to scrolling containers).

**Comment by ashetland** [comment]
@brendan-vincent-rice, anything to consider here as you wrap up pr #13906?

**Comment by matgalla** [comment]
@DintaMel What would design tokens for a selector like `::-webkit-scrollbar` look like? The semantic tokens hold values which can be used _within_ `::-webkit-scrollbar`, but I'm unfamiliar with another method of support beyond that.

@ellenupp Did you have a specific token concept in mind? Or is appropriate acceptance criteria here to ensure that all scrollbar-related definitions are inherited across components? (meaning `scrollbar-color`, `scrollbar-width`, `scrollbar-gutter` and all props defined within `::-webkit-scrollbar` selectors)

**Comment by DintaMel** [comment]
TLDR:  don't think we should create tokens for `::-webkit-scrollbar` selectors. We need clarity as to whether `scrollbar-color`, `scrollbar-width`, `scrollbar-gutter` are sufficient for @ellenupp's use case. 


> [@DintaMel](https://github.com/DintaMel) What would design tokens for a selector like `::-webkit-scrollbar` look like? The semantic tokens hold values which can be used _within_ `::-webkit-scrollbar`, but I'm unfamiliar with another method of support beyond that.

They'd still look the same but `::-webkit-scrollbar` is based on pseudo-elements.  My concerns are that they're not standard and are currently not supported in Firefox. There's a disclaimer at the top of the [MDN ::-webkit-scrollbar page](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/::-webkit-scrollbar) that says:

> Non-standard: This feature is not standardized. We do not recommend using non-standard features in production, as they have limited browser support, and may change or be removed. However, they can be a suitable alternative in specific cases where no standard option exists.

There are some differences in functionality between standard `scrollbar-` selectors and `::-webkit-scrollbar`. I mocked up a [codepen](https://codepen.io/DintaMel-the-lessful/pen/emBpLGN?editors=1000) to show some of the differences. There isn't an equivalent for `::-webkit-scrollbar-corner` and the shape of the scrollbar thumb differs.

**Comment by matgalla** [comment]
I still think the primary action item here is to ensure that components are set up to accept `scrollbar-*` and `::-webkit-scrollbar-*` definitions (if possible). It wouldn't be perfect for Windows Firefox, but ideally, at least `scrollbar-width: thin` should work as an alternative to the exact `7px` definitions Ellen mentioned above.

From there, we can consider what potential benefit scrollbar tokens might offer. I hesitate because, for example, if we were to create a `--calcite-scrollbar-size` token with a `px` value and set it up within scrollable components, it would only be usable within the pseudo-elements and not the standard `scrollbar-width` prop since it doesn't accept exact values, just `thin|auto|none`. Since the token wouldn't be able to hold a value that works for both methods, it would either be misleading or it would need to exclude Firefox Windows.

**Comment by ellenupp** [comment]
This is what scrollbars look like today in Dashboards on Mac, styled with `::-webkit-*`. **They are always visible**. 
<img width="334" height="285" alt="Image" src="https://github.com/user-attachments/assets/bf1df219-80cb-49c7-a3ea-2bacbfb31000" />

By default on Mac scrollbars appear as overlays and are not visible until the user starts scrolling. Replacing the webkit styles with `scrollbar-width: thin` and `scrollbar-color` to match dark theme, would look like:
<img width="998" height="263" alt="Image" src="https://github.com/user-attachments/assets/f9d371dd-65e0-433e-b634-38ddbc2c793f" />

Mac also has a system setting to always show scrollbars. By default (no style) they would look like this:
<img width="263" height="447" alt="Image" src="https://github.com/user-attachments/assets/7522c509-5889-453f-8797-813d4f2e50bb" />

And then “always show scrollbar” styled with `scrollbar-width: thin` and `scrollbar-color` to match dark theme:
<img width="260" height="446" alt="Image" src="https://github.com/user-attachments/assets/113ca77d-c7ff-4c13-a2be-a577d1b60530" />

Windows Chrome
Left: proposed new style
Center: default unstyled scrollbar
Right: current appearance in Dashboards
<img width="724" height="500" alt="Image" src="https://github.com/user-attachments/assets/70264cc3-035b-43d1-b338-b6c001316668" />

Same thing in Windows Firefox. Using scrollbar-width/color is an improvement!
<img width="670" height="516" alt="Image" src="https://github.com/user-attachments/assets/8025660f-fc09-4b38-b30e-54499ceddace" />

I'm generally in favor of switching to using scrollbar-width/color. It's an improvement for windows. It would be a difference for mac users who don't have scrollbars set to always show, but they are probably already used to scrolling without visible scrollbars, at least in places where scrolling is intuitive. Just one thought is always showing the scrollbars makes it clear when there is something more to see, but without that, unless the content is clipped at the bottom you might not realize that. For example in our category selector, the height is set to show exactly 6 options so you wouldn't know without hovering over it and scrolling that there could be more. 

<img width="600" height="380" alt="Image" src="https://github.com/user-attachments/assets/6b1dd48e-8d0d-4187-aaef-863c808e9f02" />

**Comment by brendan-vincent-rice** [comment]
> [@brendan-vincent-rice](https://github.com/brendan-vincent-rice), anything to consider here as you wrap up pr [#13906](https://github.com/Esri/calcite-design-system/pull/13906)?

@ashetland Apologies for the delayed response. No concerns with the Table sticky header feature. The Table's overflow/scroll approach is similar to other components that have potentially scrolling content (e.g., Panel).

**Comment by matgalla** [comment]
@ellenupp Thanks for all the examples, these are great! To confirm - is it fair to say that the newer CSS props are getting the job done? And is it appropriate to remove this acceptance criteria for this issue?
> - [ ] Scrollable components accept values defined within [`::-webkit-scrollbar` pseudo-element selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/::-webkit-scrollbar)
